lib/thor.rb

```diff
@@ -238,6 +238,9 @@ def subcommand(subcommand, subcommand_class)
         args.unshift("help") if opts.include? "--help" or opts.include? "-h"
         invoke subcommand_class, args, opts, :invoked_via_subcommand => true, :class_options => options
       end
+      subcommand_class.commands.each do |meth, command|
```

Comment:
I guess one would have to fix [Base#handle_argument_error](https://github.com/erikhuda/thor/blob/master/lib/thor/base.rb#L478) to add namespace support